hnn_core/network.py

```diff
@@ -219,47 +245,86 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
-    def _add_external_feeds(self):
-        """Legacy function, for backward compatibility with original HNN-GUI
+    def attach_drive(self, drive_name, drive_params=None,
+                     cell_specific=False):
+        """API for attaching drives according to HNN nomenclature
 
-        This function is called exactly once during initialisation
         """
-        # params of common external feeds inputs in p_common
-        # Global number of external inputs ... automatic counting
-        # makes more sense
-        # p_unique represent ext inputs that are going to go to each cell
-        self._p_common, self._p_unique = create_pext(self.params,
-                                                     self.params['tstop'])
-        self._n_common_feeds = len(self._p_common)
+        # XXX CHECKS
+        # 1. drive_name not already present
 
         # 'position' the artificial cells arbitrarily in the origin of the
         # network grid. Non-biophysical cell placement is irrelevant
         # However, they must be present to be included in gid_ranges!
         origin = self.pos_dict['origin']
 
-        # COMMON FEEDS
-        self.pos_dict['common'] = [origin for i in range(self._n_common_feeds)]
-
-        # UNIQUE FEEDS
-        for key in self._p_unique.keys():
-            # create the pos_dict for all the sources
-            self.pos_dict[key] = [origin for i in range(self.n_cells)]
+        if cell_specific:
+            drive_cell_range = range(self.n_cells)
+        else:
+            drive_cell_range = range(len(drive_params))
```

Comment:
This is quite specific to how 'common' inputs are defined: a list of parameters applied to _all_ cells.

Reply:
Hmm, I totally see why this is set up this way but the dependency on the current structure of `p_common` and `p_unique` makes me uneasy.